spec/controllers/api/predictions_controller_spec.rb

```diff
@@ -0,0 +1,118 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe Api::PredictionsController, type: :controller do
+  before(:each) do
+    controller.stub(:set_timezone)
+  end
+
+  describe 'index' do
+    before(:each) do
+      @prediction = build(:prediction)
+      @predictions = [@prediction]
+    end
+
+    context 'with valid API token' do
+      before(:each) do
+        @user = build(:user_with_email)
+        @user.stub(:api_token).and_return('token')
+        User.stub(:find_by_api_token).and_return(@user)
+        @recent = double(:recent_predictions)
+        Prediction.should_receive(:limit)
+          .with(100)
+          .and_return(double(:collection, recent: @recent))
+      end
+
+      it 'should respond with HTTP success' do
+        get :index, api_token: @user.api_token
+        response.response_code == :success
+      end
+
+      it 'should respond with JSON content type' do
+        get :index, api_token: @user.api_token
+        response.content_type == Mime::JSON
+      end
+
+      it 'should respond with predictions' do
+        get :index, api_token: @user.api_token
+        response.body.should == @recent.to_json
+      end
+    end
+
+    context 'with invalid API token' do
+      before(:each) do
+        User.stub(:find_by_api_token).and_return(nil)
```

Comment:
The test description and implementation aren't quite aligned here. The description is invalid API token so I think that there is no need to stub `find_by_api_token`. Instead just pass an invalid token in the get call. E.g. `get :index, api_toke: 'thisisnotavalidtoken'`